src/hotspot/share/utilities/vmError.cpp

```diff
@@ -424,6 +492,26 @@ static void print_oom_reasons(outputStream* st) {
   st->print_cr("# This output file may be truncated or incomplete.");
 }
 
+static void print_stack_location(outputStream* st, void* context, int& continuation) {
+  const int number_of_stack_slots = 8;
+
+  int i = continuation;
+  // Update continuation with next index before fetching frame
+  continuation = i + 1;
+  const frame fr = os::fetch_frame_from_context(context);
+  while (i < number_of_stack_slots) {
+    // Update continuation with next index before printing location
+    continuation = i + 1;
+    // decode stack contents if possible
+    const intptr_t *slot = fr.sp() + i;
+    if (is_aligned(slot, sizeof(intptr_t)) && os::is_readable_pointer(slot)) {
+      st->print("stack at sp + %d slots: ", i);
+      os::print_location(st, *(slot));
+    }
```

Comment:
Can you please print a note for unreadable slots too?

Reply:
Done.

Added early exit if sp is misaligned